plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeFileOperations.java

```diff
@@ -153,9 +165,64 @@ public void testCreateOrReplaceTableAsSelect()
                         .add(new FileOperation(DATA, "no partition", "OutputFile.create"))
                         .build());
 
+        assertFileSystemAccesses(
+                "CREATE OR REPLACE TABLE test_create_or_replace_as_select AS SELECT 2 different_col_name",
+                ImmutableMultiset.<FileOperation>builder()
+                        .add(new FileOperation(STARBURST_EXTENDED_STATS_JSON, "extendeded_stats.json", "InputFile.exists"))
+                        .add(new FileOperation(TRINO_EXTENDED_STATS_JSON, "extended_stats.json", "InputFile.newStream"))
+                        .add(new FileOperation(TRINO_EXTENDED_STATS_JSON, "extended_stats.json", "OutputFile.createOrOverwrite"))
+                        .add(new FileOperation(TRANSACTION_LOG_JSON, "00000000000000000000.json", "InputFile.newStream"))
+                        .add(new FileOperation(TRANSACTION_LOG_JSON, "00000000000000000001.json", "InputFile.newStream"))
+                        .add(new FileOperation(TRANSACTION_LOG_JSON, "00000000000000000002.json", "InputFile.exists"))
+                        .add(new FileOperation(TRANSACTION_LOG_JSON, "00000000000000000002.json", "InputFile.newStream"))
+                        .add(new FileOperation(TRANSACTION_LOG_JSON, "00000000000000000002.json", "OutputFile.createOrOverwrite"))
+                        .add(new FileOperation(LAST_CHECKPOINT, "_last_checkpoint", "InputFile.newStream"))
+                        .add(new FileOperation(DATA, "no partition", "OutputFile.create"))
+                        .build());
+
         assertUpdate("DROP TABLE test_create_or_replace_as_select");
     }
 
+    @Test
+    public void testCreateOrReplaceTableWithCheckpoints()
+    {
+        assertUpdate("CREATE TABLE test_create_or_replace_with_checkpoints (id INT, data VARCHAR) WITH (checkpoint_interval = 2)");
+
+        for (int i = 1; i <= 20; i++) {
```

Comment:
let's insert exactly once.
The commit afterwards will be the commit creating the checkpoint.

let's focus on the checkpoints, no distractions.

```
create table ... (checkpoint_interval = 2)
insert value
create or replace table ... (checkpoint_interval = 2)
check file system accesses
insert value
create or replace table ... (checkpoint_interval = 2) as select * ... 
check file system accesses
```